src/yetibot/api/aws.clj

```diff
@@ -34,90 +34,108 @@
 (defn make-aws-client
   "Returns an aws client given an aws keywordized service name (:iam, :ec2, :s3 etc.)"
   [service-name]
-  (let [client (aws/client {:api                  service-name
-                            :credentials-provider (cognitect.aws.credentials/basic-credentials-provider
-                                                    {:access-key-id     aws-access-key-id
-                                                     :secret-access-key aws-secret-access-key})
-                            :region               region})]
-    (when dev-mode?
-      (aws/validate-requests client true))
-    client))
+  (when (configured?)
```

Comment:
Added some [docs on these constraints](https://yetibot.com/dev-guide#configuration_constraints), lmk if you have feedback!

Reply:
cool thanks for the doc, I will check that out and refactor accordingly :+1:

Reply:
This is now fixed. The `region` is also `required` now since it may have some weird behaviour like the one documented [here](https://github.com/cognitect-labs/aws-api#s3-issues)